### DISCUSSION TITLE
Relays to use with Nodemcu

### DISCUSSION THREAD

**Discussion by lionlggnd** [arendst/Tasmota#18057] [Q&A] [answered]
hi, i have a nodemcu and one relay
but i have a problem: the relay is always triggered, at every gpios available and independently if toggled ON or OFF.
after various researches, i discovered out that i have a LOW LEVEL TRIGGER relay, which not works with nodemcu.
I have to find a HIGH LEVEL TRIGGER relay but i can't find it on aliexpress easily, can you provide me some links?
In alternative, can you provide me schematics and components to recreate it in gerber and print & assemble.
Thx

**Comment by sfromis** [comment]
If it is only a question of low vs high level, you can invert the pin config in Tasmota, `Relay_i` instead of `Relay`. If you are using a relay board expecting 5V signals instead of 3.3V, things may get more complex.

The Wemos D1 Mini relay shield works well with ESP8266 (or ESP32), irrespective of the form factor.
https://www.wemos.cc/en/latest/d1_mini_shield/relay.html
(Easy to find on Aliexpress or whatever)

The page there also includes a schematic for this simple design, the key is to have a suitable transistor to drive the relay, as the power available from a GPIO pin is by far not enough for that.

**Reply by lionlggnd** [reply]
i have low level 
also tried with Relay_i, but remains always active.

**Reply by lionlggnd** [reply]
i have a board that have a +5V connection (provided), GND, and input.
input can be 3,3V
i take 5V from nodemcu VU pin
GND to GND
input to D6

**Reply by sfromis** [reply]
Does the relay work as expected if you instead of a gpio simply connect the signal input to either 3.3V or GND?

**Reply by lionlggnd** [reply]
yes but if i disconnect it, it power off.
maybe low level trigger= relay triggered in LOW and HIGH gpio state because of both state in this relay can power it?

**Reply by barbudor** [reply]
@lionlggnd What is exactly your relay module ? Provide exact link

Low-level trigger will works with Relay_Inverted if and ONLY if it is a 3.3V relay. If it is an Arduino 5V relay, you need a high-level triger relay.
You already found the discussion where I explain it all

**Reply by lionlggnd** [reply]
https://it.aliexpress.com/item/1005002867727977.html?spm=a2g0o.order_list.order_list_main.75.4c933696rDXysk&gatewayAdapt=glo2ita

**Reply by barbudor** [reply]
It's a 5V relay and the description say it's low-level trigger. It won't work with a ESP which runs at 3.3V and the relay will always be ON.
With 5V relays, you must use a high-level trigger. [There](https://github.com/arendst/Tasmota/discussions/17769#discussioncomment-4771530) I linked a model that should work.

Or as pointed by sfromis the Wemos one which is explicitly designed to work with ESP

**Reply by lionlggnd** [reply]
thanks :)

**Reply by lionlggnd** [reply]
i return these relays and i'll buy high-level ones.

**Reply by satmanuk2** [reply]
@lionlggnd 
i have several of this exact relay board working fine with wemo's & nodemcu's.  it is optically isolated, so you can drive the 5V from common ground and the logic signal from a 3.3v pin (relay_i).  3.3v high keeps the relay off (no chattering)

**Reply by barbudor** [reply]
That's true for the model with 2 relays and up
But the model with 1 relay doesn't have the optocoupler

An external opto coupler could be added to solve the situation

**Reply by satmanuk2** [reply]
good point, i never noticed that (ive always purchased the 2's, 4's & 8's)

**Reply by lionlggnd** [reply]
Thanks for all yours suggestions!
Unfortunately i have returned the relays and i’m buying high-level ones. But i’ll use this suggestion as future reference.
Thanks 😁😊

**Reply by lionlggnd** [reply]
this will work? https://www.aliexpress.com/item/33012125781.html?spm=a2g0o.cart.0.0.791518fcpmEWN5&mp=1 i selected 5V high level option

**Reply by barbudor** [reply]
If we trust the description, yes

**Comment by lionlggnd** [comment, marked as answer]
from @barbudor Reply:

> It's a 5V relay and the description say it's low-level trigger. It won't work with a ESP which runs at 3.3V and the relay will always be ON. With 5V relays, you must use a high-level trigger. [There](https://github.com/arendst/Tasmota/discussions/17769#discussioncomment-4771530) I linked a model that should work.
> 
> Or as pointed by sfromis the Wemos one which is explicitly designed to work with ESP
